shelley/chain-and-ledger/formal-spec/chain.tex

```diff
@@ -316,25 +314,73 @@ \subsection{New Epoch Transition}
 
 \begin{itemize}
 \item The epoch is set to the new epoch $e$.
-\item The epoch nonce is replaced with the new one from the new epoch
-  environment.
+\item The epoch nonce is replaced with the compbination of the new one from the
+  environment and any extra entropy present in the UTxO state.
 \item The mapping for the blocks produced by each stake pool for the previous epoch
   is set to the current such mapping.
 \item The mapping for the blocks produced by each stake pool for the current epoch
   is set to the empty map.
-\item The epoch state is updated first with the rewards update \var{ru} and then
-  via the call to $\mathsf{EPOCH}$.
+\item The epoch state is updated with: first applying the rewards update \var{ru},
+  then calling the $\mathsf{EPOCH}$ transition, updating the update state, and finally
+  resetting the extra entropy mapping.
 \item The rewards update is set to \Nothing.
-\item The pool distribution is updated according to the changes in the stake pools.
+\item The new pool distribution \var{pd}' is calculated from the delegation map and
+  stake allocation of the previous epoch.
 \item A new OBFT overlay schedule is created.
 \end{itemize}
 
-The new pool distribution \var{pd}' is calculated from the delegation map and
-stake allocation of the previous epoch. It associates every stake pool key
-($\HashKey_{pool}$) to which a staking key ($\HashKey_{stake}$) delegates with
-the stake that $\HashKey_{stake}$ has. The sum of stake for all staking keys
-delegating to $\HashKey_{pool}$ then becomes the new entry of $\HashKey_{pool}$
-in the pool distribution.
+Figure~\ref{fig:funcs:new-epoch-helper} gives some helper functions for the
+$\mathsf{NEWEPOCH}$ transition.
+Three of the four functions help with the nested data structures.
+The other function, $\fun{extraEntropy}$, will return a seed value to be used as extra
+entropy for the next epoch nonce if at least five genesis keys are in agreement of the value,
+and zero otherwise.
+
+%%
+%% Figure - Helper Functions for New Epoch Rules
+%%
+\begin{figure}[htb]
+  \begin{align*}
+      & \fun{getEEnt} \in \EpochState \to \VKeyGen\mapsto\Seed \\
+      & \fun{getEEnt}~
+          (\wcard, ~\wcard,
+          ~((\wcard,~\wcard,~\wcard,~\var{H}),~\wcard,~\wcard))
+    = H
+  \end{align*}
+
+  \begin{align*}
+      & \fun{resetEEnt} \in \EpochState \to \EpochState \\
+      & \fun{resetEEnt}~
+          (\var{acnt},
+          ~\var{ss},
+          ~((\var{utxo},~\var{deposits},~\var{fees},~\wcard),~\var{dpstate},~\var{us}))
+    = \\
+      &~~
+          (\var{acnt},
+          ~\var{ss},
+          ~((\var{utxo},~\var{deposits},~\var{fees},~\emptyset),~\var{dpstate},~\var{us}))
+  \end{align*}
+
+  \begin{align*}
+      & \fun{extraEntropy} \in \EpochState \to \Seed \\
+      & \fun{extraEntropy}~\var{es} =
+        \begin{cases}
+          \eta & \text{if}~\exists!\eta(~|H\restrictrange \{\eta\}|\geq 5~)\\
```

Comment:
maybe it would be better to give the threshold in terms of the size of the genesis key mapping instead of a fixed number

Reply:
I like that idea! especially if the threshold is above .5 (which obviously it would be), then the uniqueness is obvious.